src/wallet/wallet2.cpp

```diff
@@ -6642,31 +6657,45 @@ bool wallet2::check_stake_allowed(const crypto::public_key& sn_key, const crypto
 
   /// a. Check if there is room for us
   if (full && !is_preexisting_contributor) {
-      LOG_ERROR("The service node is full.");
-      return false;
+      fail_msg_writer() << tr("This service node already has the maximum number of participants, and the specified address is not one of them.");
```

Comment:
Be mindful of the bracketing coding conventions, don't have to change it now, but worth keeping in mind. If you're developing in an isolated part of the codebase like the core tests and people aren't going to be calling into it externally from another convention that's ok. If it's integrating with the preexisting code base it should try to adhere to the conventions in the code base. Though it's easy to forget so, not too strict on it.